docs/VMR-Feature-Band-Support-Work-Plan.md

```diff
@@ -0,0 +1,146 @@
+
+Poison leak detection influences the design choices around other questions:
+* Whether artifacts provided as input to a 2xx build are separated out between 2xx PSB artifacts and 1xx artifacts or whether they are combined. If the artifact sets are provided as separate inputs, you know which are from 1xx and can designate those appropriately according to the poisoning requirements described in the previous paragraph. If they are combined, you need a way to know which came from 1xx which may require some sort of manifest file.
+* Whether a distro maintainer is required to build the *entire* 1xx branch, including tools such as Roslyn. This files from the tools would be contained in the 1xx artifacts and you'd need a way to distinguish those from the shared components so that you can allow redistribution of the shared components but disallow those from the tools.
+
+### UB Pipeline Changes
+
+#### Microsoft Build Legs
+
+Some of the Microsoft build legs will need to be trimmed from the 2xx branch. Specifically, there are short stack and cross-OS DAC builds that only build the runtime repo. These would not be relevant for the 2xx branch and should be excluded in the pipeline.
+
+Darc publishing will need to be enabled at the VMR level to allow the packages produced from the 1xx build to flow and be consumed by the sources in the 2xx branch.
+
+#### Source-Only Build Legs
+
+Given that the packages outputted from the 1xx branch are needed in order to source build the 2xx branch, changes will be needed to the pipelines. In order to make the build efficient, we can collect the output of the 1xx branch's official build and feed that in as input to the 2xx branch build. There's no need to build the 1xx branch again from source since it's already been done.
+
+In order to do this, there needs to be a mechanism to publish the source-built packages from a 1xx official build so that they can be consumed by a 2xx build. There are a variety of options that can be considered:
+* Publish each source-built NuGet package to a NuGet feed (probably specific to source build). These would then be downloaded all together by using [NuGet search functionality](https://learn.microsoft.com/nuget/reference/nuget-client-sdk#search-packages) by filtering on the package version.
+* Publish each source-built artifacts tarball to blob storage. This would then be downloaded by constructing a URL that contains the necessary version.
```

Comment:
The benefit of using a tarball is that we get the benefit or storing other files besides just packages. For example, we may end up needing the VerticalManifest.xml file in the tarball so that we know which repo each package came from. That's necessary for source build poison leak detection where we would allow packages from shared components to be "leaked" but nothing else.

Reply:
Yes, I think it makes a ton of sense to use this method. This also reduces the possibility of SB packages with the same identities as MS's but different content leaking somewhere.

Reply:
We'd probably store it in https://ci.dot.net. How does the retention work for other archive files that get published there? Do they just remain there indefinitely? Do we see any issue with doing the same with this scenario? The important thing we want to allow from a retention policy is that any person should be able to go to a past release tag from a 2xx release and still be able to build it with access to the 1xx artifacts associated with it at that time.

Reply:
Retained forever.